lib/util/http/host-rules.ts

```diff
@@ -162,6 +163,28 @@ export function applyHostRules<GotOptions extends HostRulesGotOptions>(
   if (!hasProxy() && foundRules.enableHttp2 === true) {
     options.http2 = true;
   }
+
+  if (is.nonEmptyString(foundRules.httpsCertificateAuthority)) {
+    options.https = {
+      ...(options.https ?? {}),
+      certificateAuthority: foundRules.httpsCertificateAuthority,
+    };
+  }
+
+  if (is.nonEmptyString(foundRules.httpsPrivateKey)) {
```

Comment:
`mTLS` always requires a private key and cert. If split like here, invalid configurations may be supplied.

Reply:
This test is more focusing on whether the correct options has been passed down to the `got` options.

Reply:
So it should be part of config/validation?

Reply:
I tried it in my local got test example. it doesn't seem like `got` enforces both `key` and `cert` to be presented at the same time. I wonder if there are other use case that beyonds `mtls` we don't know just yet that needs this kind of config to happen. I would suggest leave it as is.